### PR TITLE
overview: update with example from api repo

### PR DESCRIPTION
**Pull request by spring4175**

### CODE REVIEW

File: src/overview.md [hunk 1]
```diff
@@ -46,45 +46,83 @@ Below is a quick example of a program printing "Pong!" when a ping command comes
...
-use std::{
-    env,
-    error::Error,
+use std::{env, error::Error};
+use tokio::stream::StreamExt;
+use twilight::{
+    cache_inmemory::{EventType, InMemoryCache},
+    gateway::{cluster::{Cluster, ShardScheme}, Event},
+    http::Client as HttpClient,
+    model::gateway::GatewayIntents,
```
Comment (Erk-) [inline src/overview.md]: Should this be individual crates?

Comment (spring4175) [reply src/overview.md]: Resolved

**Review by Erk- — APPROVED** [review]

File: src/overview.md [hunk 2]
```diff
@@ -46,45 +46,81 @@ Below is a quick example of a program printing "Pong!" when a ping command comes
...
-
-    while let Some(event) = events.next().await {
-        match event {
-            Event::MessageCreate(msg) => match parser.parse(&msg.content) {
-                Some(Command { name: "ping", .. }) => println!("Pong!"),
-                _ => {},
-            },
-            // More events here...
-            _ => {},
+    // This is also the default.
```
Comment (7596ff) [inline src/overview.md:60]: "also" - this is the first comment in this example. reword?

File: src/overview.md [hunk 3]
```diff
@@ -46,45 +46,81 @@ Below is a quick example of a program printing "Pong!" when a ping command comes
...
+    let scheme = ShardScheme::Auto;
+
+    let cluster = Cluster::builder(&token)
+        .shard_scheme(scheme)
+        // Use intents to only listen to GUILD_MESSAGES events
```
Comment (7596ff) [inline src/overview.md:65]: comment does not match code

File: src/overview.md [hunk 4]
```diff
@@ -46,45 +46,81 @@ Below is a quick example of a program printing "Pong!" when a ping command comes
...
+        .intents(Some(
+            GatewayIntents::GUILD_MESSAGES | GatewayIntents::DIRECT_MESSAGES,
+        ))
+        .build()
+        .await?;
+
+    // Start up the cluster
```
Comment (7596ff) [inline src/overview.md:72]: need to decide whether to use periods here

File: src/overview.md [hunk 5]
```diff
@@ -46,45 +46,81 @@ Below is a quick example of a program printing "Pong!" when a ping command comes
...
+    let cluster_spawn = cluster.clone();
+
+    tokio::spawn(async move {
+        cluster_spawn.up().await;
+    });
+
+    // The http client is seperate from the gateway,
+    // so startup a new one
```
Comment (7596ff) [inline src/overview.md:80]: "a new one" doesn't really make sense here because the gateway's client is hidden in impl. clarify?

**Review by 7596ff — CHANGES_REQUESTED** [review]

**Pull request merged**